### ISSUE TITLE
Rendered markdown is incorrect when table includes empty cells

### ISSUE THREAD

**Issue by grantwinney** [jmalarcon/markdowntables#1]
Hi, thanks for uploading this. It's nice to have a place to suggest fixes.

I noticed a bug with the rendered markdown that existed in the original page, albeit slightly different.

In your script, an empty cell like this:

```html
<table>
    <tr><td>1</td><td></td><td>3</td></tr>
    <tr><td>1</td><td>2</td><td>3</td></tr>
</table>
```

Generates markdown that is missing that column.

```
|1|3|
|1|2|3|
```

The original one had the same problem, but inserting a blank space like `<td> </td>` caused it to generate the correct markdown:

```
|1||3|
|1|2|3|
```

I'm assuming it has to do with this block:

```javascript
// if there is a thead we append
$(html).find('thead > tr > td').each(function() {
	if(fixText($(this).text()) != ""){
		table_header = table_header + fixText($(this).text()) + "|";
		table_header_footer = table_header_footer + "--- |";
		table_header_found = true;
	}
});
```

Which may have looked like this in the original:

```javascript
// if there is a thead we append
$(html).find('thead > tr > td').each(function() {
	if($(this).text() != ""){
		table_header = table_header + $(this).text() + "|";
		table_header_footer = table_header_footer + "--- |";
		table_header_found = true;
	}
});
```

Do you think this would work? It allows for an empty cell.

```javascript
// if there is a thead we append
$(html).find('thead > tr > td').each(function() {
        var fixedText = fixText($(this).text());
	if($(this).text() == "" || fixedText != ""){
		table_header = table_header + fixText($(this).text()) + "|";
		table_header_footer = table_header_footer + "--- |";
		table_header_found = true;
	}
});
```

Or since the original one seemed to generate the correct markdown even if the table cell had multiple spaces in it, perhaps just don't "fix" those:

```javascript
function fixText(text){
    //Remove tabs (it's HTML, sothey don't affect, but in Markdown, they do.)
    return text.trim().replace('\t','');
}
```

**Comment by jmalarcon** [comment]
Hi Grant:

You're right. There's a bug with this. Thanks for opening the issue and letting me know.

I've just fixed it and pushed the new version, that is now live. I've removed the `if` altogether and simply fixed the value when added to the Markdown. BTW, I've seen that the original code is online again [here](http://markdowntables.mrvautin.com/). It has the same problem with empty cells, that now my version has fixed.

I've also added two new features:

- Add support for headless tables. If a table lacks a header an empty it's added, since most Makdown processors don't support headless tables
- Add a check to test that all the rows have the same number of cells (including the header)

HTH
